src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -30712,20 +30712,567 @@ public interface I1
+    {
+        I1 x = new Test();
+        x.I1();
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe,
+                                                 parseOptions: TestOptions.Regular);
+
+            compilation1.VerifyDiagnostics();
+
+            CompileAndVerify(compilation1, expectedOutput: "Test.I1");
+        }
+
+        [Fact]
+        public void Constructors_09()
+        {
+            var source1 =
+@"
+interface I1
+{
+    void I1();
+    static I1()
+    {
+        System.Console.WriteLine(""I1..cctor"");
+    }
+    static void M1()
+    {
+        System.Console.WriteLine(""I1.M1"");
+    }
+}
+
+class Test : I1
+{
+    void I1.I1()
+    {
+        System.Console.WriteLine(""Test.I1"");
+    }
+
+    static void Main() 
+    {
+        I1.M1();
+        I1 x = new Test();
+        x.I1();
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe,
+                                                 parseOptions: TestOptions.Regular);
+
+            compilation1.VerifyDiagnostics();
+
+            CompileAndVerify(compilation1, expectedOutput:
+@"I1..cctor
+I1.M1
+Test.I1
+");
+        }
+
+        [Fact]
+        public void Constructors_10()
+        {
+            var source1 =
+@"
+interface I1
+{
+    void I1()
+    {
+        System.Console.WriteLine(""I1.I1"");
+    }
+
+    static I1()
+    {
+        System.Console.WriteLine(""I1..cctor"");
+    }
+    static void M1()
+    {
+        System.Console.WriteLine(""I1.M1"");
+    }
+}
+
+class Test : I1
+{
+    static void Main() 
+    {
+        I1.M1();
+        I1 x = new Test();
+        x.I1();
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe,
+                                                 parseOptions: TestOptions.Regular,
+                                                 targetFramework: TargetFramework.NetStandardLatest);
+
+            compilation1.VerifyDiagnostics();
+
+            CompileAndVerify(compilation1, expectedOutput: !CoreClrShim.IsRunningOnCoreClr ? null :
+@"I1..cctor
+I1.M1
+I1.I1
+",
+                verify: VerifyOnCoreClr);
+        }
+
+        [Fact]
+        public void Constructors_11()
+        {
+            var source1 =
+@"
+interface I1
+{
+    void I1()
+    {
+        System.Console.WriteLine(""I1.I1"");
+    }
+}
+
+class Test : I1
+{
+    static void Main() 
+    {
+        I1 x = new Test();
+        x.I1();
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe,
+                                                 parseOptions: TestOptions.Regular,
+                                                 targetFramework: TargetFramework.NetStandardLatest);
+
+            compilation1.VerifyDiagnostics();
+
+            CompileAndVerify(compilation1, expectedOutput: !CoreClrShim.IsRunningOnCoreClr ? null : "I1.I1", verify: VerifyOnCoreClr);
```

Comment:
The runtime has facilities for dealing with cyclic static constructors on classes / structs. The behavior should be the same for interfaces (largely undefined ordering but executes without crashing). Think we should add one sanity test here just to ensure they have hooked up this code path.

Reply:
> The runtime has facilities for dealing with cyclic static constructors on classes / structs. The behavior should be the same for interfaces (largely undefined ordering but executes without crashing). Think we should add one sanity test here just to ensure they have hooked up this code path. 

Sure, let's discuss offline what you have in mind in more details.

---
In reply to: [260422959](https://github.com/dotnet/roslyn/pull/33625#discussion_r260422959) [](ancestors = 260422959)